edx_lint/pylint/feature_toggle_check.py

```diff
@@ -0,0 +1,186 @@
+"""Pylint plugin: checks that feature toggles are properly annotated."""
+
+import re
+
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+
+from .common import BASE_ID, check_visitors
+
+
+def register_checkers(linter):
+    """Register checkers."""
+    linter.register_checker(FeatureToggleChecker(linter))
+
+class AnnotationLines(object):
+    """
+    StringLines provides utility methods to work with a string in terms of
+    lines.  As an example, it can convert an index into a line number or column
+    number (i.e. index into the line).
+    """
+
+    # Regex searches for annotations like: [    # .. ]
+    _ANNOTATION_REGEX = re.compile(r'[\s\S]*#[\s\S]*?\.\.[\s\S]*?feature_toggle')
+
+    def __init__(self, module_node):
+        """
+        Arguments:
+            module_node: The visited module node.
+        """
+        module_as_binary = module_node.stream().read()
+        self._string = module_as_binary.decode("iso8859-1")
+
+        self._line_start_indexes = self._process_line_breaks(self._string)
+        # this is an exclusive index used in the case that the template doesn't
+        # end with a new line
+        self.eof_index = len(self._string)
+
+    def is_line_annotated(self, line_number):
+        """
+        Checks if the provided line number is annotated.
+        """
+        if line_number < 1 or self.line_count() < line_number:
+            return False
+
+        return bool(
+            self._ANNOTATION_REGEX.match(self._line_number_to_line(line_number))
+        )
+
+    def line_count(self):
+        """
+        Gets the number of lines in the string.
+        """
+        return len(self._line_start_indexes)
+
+    def _process_line_breaks(self, string):
+        """
+        Creates a list, where each entry represents the index into the string
+        where the next line break was found.
```

Comment:
This sounds like a complex thing that could be done more easily by just keeping a list whose elements were the lines of source text.

Reply:
As noted, I just grabbed old code that was doing this from the XSS linter.  I agree something simpler would work.